lib/output.js

```diff
@@ -597,6 +598,12 @@ function OutputStream(options) {
             return true;
         }
 
+        if (output.option('webkit')
+            && output.parent(0) instanceof AST_PropAccess
+            && output.parent(1) instanceof AST_Assign) {
```

Comment:
> `function(){1+1}.a` works fine under Safari 5

...and presumably under `phantomjs`.

That's interesting. It's so rare and obscure I wouldn't make the distinction between LHS and RHS - especially in light of the fact that it is behind a "webkit" option.

In any case the patch above does not look at the parent `expression` leading to unnecessary parens:
```
$ echo 'console.log(a[function(){1 + 1}] = 1);' | bin/uglifyjs -b webkit,beautify=false
console.log(a[(function(){1+1})]=1);
```